core/templates/dev/head/pages/exploration_player/AudioControlsDirective.js

```diff
@@ -56,20 +59,37 @@ oppia.directive('audioControls', [
               '/icons/rewind-five.svg'));
 
           $scope.isAudioAvailableInCurrentLanguage = function() {
-            return Boolean(getCurrentAudioTranslation());
+            return Boolean(getAudioTranslationInCurrentLanguage());
           };
 
           $scope.doesCurrentAudioTranslationNeedUpdate = function() {
-            return getCurrentAudioTranslation().needsUpdate;
+            return getAudioTranslationInCurrentLanguage().needsUpdate;
           };
 
-          $scope.playPauseAudioTranslation = function() {
-            // TODO(tjiang11): Change from on-demand loading to pre-loading.
+          $scope.onSpeakerIconClicked = function() {
+            var audioTranslation = getAudioTranslationInCurrentLanguage();
+            if (audioTranslation) {
+              if (!AudioPreloaderService.hasPreloadedLanguage(
+                    getCurrentAudioLanguageCode()) &&
+                    !isCached(audioTranslation)) {
+                AudioPreloaderService.excludeFile(
```

Comment:
Ah okay, yeah that's what I suggested in the earlier lost comment--will add that